studio/components/layouts/SQLEditorLayout/SQLEditorMenu.tsx

```diff
@@ -110,9 +105,20 @@ const DropdownMenu = observer(({ tabInfo }: { tabInfo: QueryTab }) => {
         buttonLabel="Confirm"
         visible={deleteModalOpen}
         onSelectConfirm={async () => {
-          await contentStore.del(id)
-          await contentStore.load()
+          // TODO: remove this hack
+          // it's needed because the onClick handler from the menu item registers here, instead
+          // of on just the menu item. Meaning that the selectTab runs, causing an error.
+          // Once a fix is made in supabase/ui this can safely be removed.
+          await new Promise((resolve) => setTimeout(resolve, 10))
```

Comment:
Update: I'm using a better workaround in the form of `event.preventDefault()` and `event.stopPropagation()`. It would be a nice to have if supabase/ui handled this for us but low on the priorities list!